Feat/header

karstenlenz opened the pull request:
Please review my code.

closes #35

maximiliangustmann commented on src/App.js:
```diff
@@ -12,12 +12,62 @@ import JournalForm from './Journal/JournalForm'
+import Header from './common/Header'
 
 export default function App() {
   const [journalEntries, setJournalEntries] = useState(
     loadLocally('journalEntries') ?? []
   )
 
+  useEffect(() => saveLocally('journalEntries', journalEntries), [
+    journalEntries,
+  ])
```
You could move this easily into a custom hook, your App.js looks a bit cleaner this way. Would also move line 71-82 in the same custom hook and call it useJournalEntries.

karstenlenz replied:
Good idea! I added the custom hook!

maximiliangustmann commented on src/App.js:
```diff
@@ -12,12 +12,62 @@ import JournalForm from './Journal/JournalForm'
+  return (
+    <>
+      <Switch>
+        <Route exact path="/"></Route>
+        <Route path="/">
+          <Header />
+        </Route>
```
Just curios, does this make your header appear on every page execpt your homepage? Otherwise I would not understand the two path's to "/"

karstenlenz replied:
Yes, exactly. I couldn't find another way to put the header on every page except for the homepage.

maximiliangustmann commented on src/common/Header.js:
```diff
@@ -0,0 +1,32 @@
+        altText="zurück"
+        onClick={() => history.goBack()}
+      />
+      <img src="/img/logo.svg" alt="Youlity logo" />
```
With react router you could also import your image as react component.

karstenlenz replied:
Yes, but I think if I import it as a component I can't set an alt text.

maximiliangustmann commented on src/common/IconButton.js:
```diff
@@ -0,0 +1,24 @@
+}
+
+const ButtonContainer = styled.button`
+  width: 40px;
+  height: 40px;
```
Should your IconButton have the 40x40 size on every display size? Otherwise I would use a more responsive size, like you did at other parts of your code.

karstenlenz replied:
Yes, the size is fixed for every display size. I don't think it makes sense to have huge Icons on Desktop screens.

maximiliangustmann reviewed (commented):
This is honestly my favorite W4 App so far and your code shows why.

karstenlenz commented:
> This is honestly my favorite W4 App so far and your code shows why.

Thank you for your review!

carolinwagner commented:
QA Feedback:

Your header works great - I can go back and go to "home" from every single page. 
Some minor thoughts: 

- on page "Schritt 1: Bedürfnisse" the "go back" icon and the "home" icon are leading to the same page. Maybe you don't need both icons on this page - I'd understand if you keep them due to consistency, but as a user I was wondering what the difference is
- while answering your questions, the "go back" icon goes back to "Schritt 2: Fragebogen". As a user I might think, the "go back" icon goes back to the previous question 
- when clicking the answer possibilities a little bit faster than usual, every other question is skipped and automatically answered with the answer possibility clicked before (I hope you can reproduce this)

karstenlenz commented:
> QA Feedback:
> 
> Your header works great - I can go back and go to "home" from every single page.
> Some minor thoughts:
> 
>     * on page "Schritt 1: Bedürfnisse" the "go back" icon and the "home" icon are leading to the same page. Maybe you don't need both icons on this page - I'd understand if you keep them due to consistency, but as a user I was wondering what the difference is
> 
>     * while answering your questions, the "go back" icon goes back to "Schritt 2: Fragebogen". As a user I might think, the "go back" icon goes back to the previous question
> 
>     * when clicking the answer possibilities a little bit faster than usual, every other question is skipped and automatically answered with the answer possibility clicked before (I hope you can reproduce this)

Thanks for testing! 
- I will leave the back and home button in the header, even if they point to the same page. Otherwise it might be confusing. I think this is standard behavior on most apps or websites.
- You are right, this would be annoying. I added a task to my backlog to show a warning if the user wants to go back. 
- Thanks for catching that! I fixed it by deactivating the buttons for 100ms after an answer.
